homeassistant/components/webostv/__init__.py

```diff
@@ -1 +1,205 @@
 """Support for WebOS TV."""
+import asyncio
+import logging
+
+from aiopylgtv import PyLGTVCmdException, PyLGTVPairException, WebOsClient
+import voluptuous as vol
+from websockets.exceptions import ConnectionClosed
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_CUSTOMIZE,
+    CONF_FILENAME,
+    CONF_HOST,
+    CONF_ICON,
+    CONF_NAME,
+    ENTITY_MATCH_ALL,
+    EVENT_HOMEASSISTANT_STOP,
+)
+import homeassistant.helpers.config_validation as cv
+
+DOMAIN = "webostv"
+
+CONF_SOURCES = "sources"
+CONF_ON_ACTION = "turn_on_action"
+CONF_STANDBY_CONNECTION = "standby_connection"
+DEFAULT_NAME = "LG webOS Smart TV"
+WEBOSTV_CONFIG_FILE = "webostv.conf"
+
+SERVICE_BUTTON = "button"
+ATTR_BUTTON = "button"
+
+SERVICE_COMMAND = "command"
+ATTR_COMMAND = "command"
+
+CUSTOMIZE_SCHEMA = vol.Schema(
+    {vol.Optional(CONF_SOURCES): vol.All(cv.ensure_list, [cv.string])}
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.All(
+            cv.ensure_list,
+            [
+                vol.Schema(
+                    {
+                        vol.Optional(CONF_CUSTOMIZE, default={}): CUSTOMIZE_SCHEMA,
+                        vol.Optional(
+                            CONF_FILENAME, default=WEBOSTV_CONFIG_FILE
+                        ): cv.string,
+                        vol.Optional(CONF_HOST): cv.string,
+                        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                        vol.Optional(CONF_ON_ACTION): cv.SCRIPT_SCHEMA,
+                        vol.Optional(
+                            CONF_STANDBY_CONNECTION, default=False
```

Comment:
Some TV models have a "Standby+" option which causes them to go to some kind of standby/suspend mode rather than powering off.  In this state the tv will not respond to websocket ping commands, however it will keep the connection and subscribed state updates alive and immediately send a state update on power on.  This option tells the web os client not to disconnect on power off and to suppress the usual ping pong connection checks when the TV is off.
(The advantage of this is that if the TV is switched back on outside of home assistant, e.g. with the remote, home assistant will be updated immediately, since the first thing the TV will do is send a state update.)

Reply:
Is it impossible for the client to detect if the TV supports the standby option?

Reply:
Whether the TV supports the option could maybe be inferred from the model number, but in any case the option is user-configurable in the TV UI, and the appropriate setting would depend on how the user has set it, which is not possible for the client to detect at the moment.

Reply:
Ok, then we keep this option here.